### PR TITLE
tf: tfprint method

### PR DESCRIPTION
**Pull request by torbjorgen**
Signed-off-by: Tor Björgen <tor.bjorgen@scilifelab.se>

### CODE REVIEW

File: src/rdds/lib/tf/tfprint.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from typing import Union
+import keras.engine.keras_tensor
+import tensorflow as tf
+
+TensorTypes = Union[tf.Tensor, tf.RaggedTensor, keras.engine.keras_tensor.KerasTensor]
+
+
+def tfprint(tensor: TensorTypes,
```
Comment (henrikstranneheim) [inline src/rdds/lib/tf/tfprint.py]:
```suggestion
def tf_print(tensor: TensorTypes,
```

File: src/rdds/lib/tf/tfprint.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            summarize: int = -1) -> TensorTypes:
+    """
+    Prints a Tensor or KerasTensor when executed inside the computational graph.
+    :param tensor: The tensor to be printed
+    :param message: A text message
+    :param summarize: -1 prints content of all tensor, otherwise n samples
+    :return: The printed tensor, make sure to use it in the graph to have this method called.
+    """
+    tensor = tf.keras.backend.print_tensor(tensor, message=message, summarize=summarize)
+    return tensor
```
Comment (henrikstranneheim) [inline src/rdds/lib/tf/tfprint.py]: Doesn't feel intuitive to return something from a print function. Would be good to rename the function or reconsider the return.

**Review by henrikstranneheim — APPROVED** [review]
👍

File: src/rdds/lib/tf/tfprint.py (hunk 2 above)
Comment (torbjorgen) [reply src/rdds/lib/tf/tfprint.py]: I agree, the API is quite odd. I renamed the method to `print_tensor_op` which is more in line with the actual method of performing the print call. An `op` in tensorflow lingo is considered a method that performs work on some tensor, returning the instance of the processed tensor on finish.

**Pull request merged**